[READY] Reinforced windows are harder to get through

actioninja opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

full tile Reinforced windows now require a melee item over 11 damage to damage, have 150 health, and 90 melee armor. This is approximately 2 minutes of sustained beating with a toolbox.
full tile reinforced Plasmaglass windows now require a melee item over 21 damage to damage, have 200 health, and 95 melee resistance. This is approximately 3 minutes of sustained beating with a toolbox.
dir windows have been given the appropriate halved health values.

Current deconstruction steps:
1. heat one way screws (welder, 18 seconds, harm intent required)
2. unscrew one way screws (screwdriver, 8 seconds)
3. pop out panel (crowbar, 5 seconds)
4. cut connecting bars (wirecutter, 3 seconds)
5. unbolt frame (wrench, 5 seconds)
theoretical fastest time: 39 seconds
more practical time: 45 seconds

if the bolts are heated and then not unscrewed within 30 seconds they cool off again, and must be heated again

Construction is the same as before except the crowbar levering step takes 10 seconds instead of 3.


## Why It's Good For The Game

Getting into places you shouldn't be is currently way too easy.
This change forces you to hack in, disassemble the window, or target a weak point like a windoor.
At the moment, it is possible to smash down what is intended to be a secure window with an item you get in your bag at roundstart for free. This is stupid. It should require some planning or at least thinking to be able to make your way in places you should not be.

Oranges has talked in the past about departments being designed like impenetrable fortresses with shutters everywhere, and this happens because reinforced windows are far too weak and unreliable to be able to have as effective defense of any department against even underequipped and underprepared people. This leads to the introduction of shutters which are too strong and difficult to deal with in most cases.

This is part of some plans to make departments more secure overall while still providing ample avenues for determined or properly prepared individuals able to get in if they want to. Despite having more plans along with this, this is a standalone change as is. This is not a "this will only work with later changes I'm totally going to do" situation

## Changelog
:cl:
balance: Reinforced windows now require a melee item over 11 damage to damage, have 150 health, and 90 melee armor. This is approximately 2 minutes of sustained beating with a toolbox.
balance: Plasmaglass windows now require a melee item over 21 damage to damage, have 200 health, and 90 melee resistance. This is approximately 3 minutes of sustained beating with a toolbox.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

zxaber commented:
Screwdriver and crowbar buff.

Honestly, the monkeys that just smash their way in are silly, anyway.

optimumtact commented:
you probably need to make the reinforced window deconstruct recipe a little more than screwdriver, crowbar too then

zxaber commented:
That presumably would make replacing them more tedious too, for better or worse.

Plasma glass windows should have more steps though, similar to R-walls vs normal walls.

actioninja commented:
I'm aware of that this will make having tools more important, and that is not unintentional.

optimumtact commented:
zxaber, well yes, although construction can always be faster than deconstruction with both reinforced walls and glass, plus RPD's as well.

Not a huge loss I would say.

Whoneedspacee commented:
[relevant](https://www.youtube.com/watch?v=x_CiLZY-HGw)

TetraK1 commented:
> [relevant](https://www.youtube.com/watch?v=x_CiLZY-HGw)

space glass > earth glass

also want windoors stronger as well, not a fan of grey lying under flaps toolboxing windoors

optimumtact commented:
did you even see the video? dude was going at that last window with everything and couldn't do more than crack it

TetraK1 commented:
no, i'm an impatient child and only watched to the end of the second last one

Dawson1917 commented:
perma is now gay baby jail why are you ruining my hugbox like this


I feel these numbers are a bit too high, since almost nobody is able to get their hands on a weapon able to do above 21 damage without buying an esword or fetching one of the two fireaxes on the station. And that's only to break a window in the HoPline, or RnD, or wherever else. Just about every room on the station has reinforced windows except public areas like primary tool storage. These places really need to have their windows replaced to not make every department an impenetrable fortress unless you fetch a screwdriver and crowbar (don't make reinforced windows r-walls V2). Why do low-risk areas like genetics or cargo lobby or the lawyer's office or the outer HoPline or the corporate showroom or the library or the theater or the holodeck have to be reinforced with ballistic-grade glass?
Even lowering to 15 damage restricts what weapons can break windows to the level of welders and surgical saws. Ideally, though, someone would implement a system of brute-impact objects like pickaxes or sledgehammers(when?) have a greater effect on smashing a window than poking it with a welder or knife

Whoneedspacee commented:
who would win

dawson or a screwdriver and a crowbar

Dawson1917 commented:
> 
> 
> who would win
> 
> dawson or a screwdriver and a crowbar
> 
> did you know prisons have glass on their doors

If this gets merged reinforced windows are going to get a much more tedious deconstruct time.

knacker48 commented:
Won't this mean that, in those times when there's no power, people getting off the arrivals shuttle will be stuck? And before you say 'but there's a crowbar in the emergency toolbox', half the time people either take it for a free crowbar or worse they use it to open the airlocks then close them again to stop the station from getting depressurized

fludd12 commented:
Maybe make it only do, like, a quarter of its damage if it is below that damage amount, rather than no damage at all? While I support making reinforced windows more robust, if you go at glass with a solid object enough, it doesn't matter how reinforced it is, it _will_ eventually crack. It'll make deconstruction a lot faster than just smashing it down, but still allow players to break in if they've got a lot of time on their hands (such as being in perma).

Akrilla commented:
21 puts it just above the spear, which is what, only fireaxe?

Akrilla commented on code/game/objects/obj_defense.dm:
```diff
@@ -22,6 +22,8 @@
 
 ///returns the damage value of the attack after processing the obj's various armor protections
 /obj/proc/run_obj_armor(damage_amount, damage_type, damage_flag = 0, attack_dir, armour_penetration = 0)
+	if(damage_flag == "melee" && damage_amount < damage_deflection)
```
Would this mean that a shotgun pointblank would do nothing?

nemvar replied:
Shotgun has damage_flag == "bullet".

Akrilla replied:
Yeah lmao read it wrong, thought only melee and damage over damaged

CPTANT commented:
21 is the same as required for doors, which is way too much.

Reinforced glass already takes a decent amount of time to bash down, which is exactly what it should do. It's easy to bash down when nobody is looking but it's hard to do unnoticed.

nicbn commented:
Good change.

I feel like reinforced windows should be the "wall-strength" variant of a window (a transparent wall, if you wish), but currently they are just a window that takes more hits.

I feel like doors should be stronger too (gamedesign-wise, doors shouldn't be a

Dawson1917 commented:
> I feel like reinforced windows should be the "wall-strength" variant of a window (a transparent wall, if you wish), but currently they are just a window that takes more hits.

That's what plasmaglass windows are for though. They take forever to smash down even with an armblade or esword, let alone the much weaker objects crew have access to. Reinforcing windows with plasmaglass is a good project for engineers to do in their free time. Making regular reinforced windows borderline invincible to 95% of objects or weapons is... wow

Shapsy commented:
This would make it almost impossible to get around the station in an emergency, I can't think of a single weapon general crew would have access to that's over 21 force other than the fireaxe. Breaking reinforced windows is very loud and time consuming, and security can just shoot you through the glass if they see you doing it anyway. The emergency oxygen tank you mention would take 25 straight hits in a row to shatter the window, which is probably 10-15x slower than just pulling the window out, which seems like a pretty fair trade to me (not that I'd ever think to use an emergency oxy tank to break a window)

Desperately breaking a window with a nearby toolbox once you see plasma start coming in through the vents and the doors bolt shut is prime SS13 in my mind, and it'd suck if you just had to sit there and die because you didn't go buy a screwdriver and crowbar roundstart as a doctor

Whoneedspacee commented:
At the very least breaking the window should be more difficult than deconstructing it properly, and that isn't the case right now. It's pretty silly that reinforced windows are basically seen as the base level of windows because someone with a toolbox can smash them down in about 10 seconds. There's literally no point to even making normal windows.

nicbn commented:
> That's what plasmaglass windows are for though. 

I feel like plasmaglass should be anti-radiation, high-temperature-resistant reinforced glass (even though reinforced glass is already resistant to high temperature, I don't think it should be).

> This would make it almost impossible to get around the station in an emergency...

That's a station design issue (that's actually caused by the fact that windows are so weak). The "internal" areas should be using normal windows, not reinforced. But the reinforced windows are currently weak, so it's mapped on just to thoughen up the area a bit, or whatever.

> ...security can just shoot you through the glass if they see you doing it anyway.

Well, first off that's relying on security to balance a mechanic design - that's like saying eswords should be 1-hit-kill because security can shoot you to stop you anyway.

Second, I feel like that should be the case with regular windows. Maybe make them a little (just a little) thougher so that security can come and have a chance to get the criminal near the crime scene (unless they use a fire axe which should be always 1-hit-destroy).

Swagile commented:
This will fuck over crew a lot more then you think.

As stated above, in emergencies, you might as well suicide unless you, round start every round, rush tool storage and buy tools. Emergency toolboxes don't come with screwdriver + crowbar, and like 80% of the station is reinforced glass.

This needs a lot of other changes before this is accepted, such as lowering r-window dependency (removing it from most non-secure areas), giving emergency toolboxes the tools needed to open r-windows, or adding other escape routes from departments such as more disposals / plastic flaps.

Anything but this; this feels like a extremely rushed change without much thought put into its after affects, which is mostly harming crew immensely while profiting antagonists as most antags can get their hands on something that is over 21 damage.

Akrilla commented:
I definitely don't think this is a bad change, and its impact won't be massive, I just don't think they should basically be indestructible to basically anything besides tool deconstruction and explosions.

Krysonism commented:
Just give them a lot more hitpoints or some damage reduction. Throwing spears to open lockers is an extremely hacky mechanic that makes no sense.

If anything spears and knives should be worse at breaking down windows while heavy tools and mining equipment should do extra damage.

I think that toolboxing down a reinforced window should be too tedious to be bothered with in most circumstances, but not impossible.

Tlaltecuhtli commented:
why dont we add different kinds of glasses?

SkeletalElite commented:
As Krysonism said, I think increasing the health or giving them damage resistance is a better solution. Make it strong enough that basing them down is harder than just deconstructing them if you want to get somewhere but still actually possible to do in an emergency.

CPTANT commented:
Why does everyone keep saying it isn't harder to bash them down then deconstruct them?

It takes way longer to bash a window dozens of times than it does to screwdirver-crowbar-screwdriver

nemvar commented:
What the fuck does the crew even have with more than 21 damage? Fireaxe, chainsaws, thrown spears and sharpened nullrod. This is a poor solution to a complex problem.

Shapsy commented:
Also why are people saying windoors are a weak point when they have 1.5x more effective health than reinforced windows (150hp for a windoor vs 50hp and 50% melee resist for a r-window) and take more than two steps to unanchor

nemvar commented:
The real weak points of rooms are the god damn walls. Only two steps to deconstruct and can get destroyed instantly if you have the right tool.

Swiftkor commented:
I support this, very sick of assistants breaking windows for the lulz with little effort. Also might make medbay storage window somewhat of a deterrent (not likely)

MCterra10 commented:
+1, sick of clown toolboxing my windows as chemist when I don't give him lube

actioninja commented:
I think there's some validity to concerns of getting GBJed in places, but I don't think that's necessarily a bad thing.

Dawson1917 commented:
> 
> 
> I think there's some validity to concerns of getting GBJed in places, but I don't think that's necessarily a bad thing.

The majority of places on the station shouldn't be GBJ the moment a door closes behind you. Why are 90% of all windows ballistics-grade reinforced glass, 5% plasma glass, and 5% actual windows? They can be more difficult to break, by all means, but it should still be possible to get out of a room through a bloody piece of glass. 90% of station windows now requiring a fireaxe or esword to break through is ridiculous considering you're most likely to be smashing a window into chemistry for meds/cures in the absence of chemists or breaking into RnD because they are literal troglodytes who have neither researched nor upgraded anything on the station

MCterra10 commented:
@Dawson1917 in either case, is it REALLY that hard to take apart the window rather than smash it down? You don't have to destroy everything you know

CPTANT commented:
> 
> 
> +1, sick of clown toolboxing my windows as chemist when I don't give him lube

Conflict is part of the game and you can just put down the shutters anyway.

Dawson1917 commented:
> 
> 
> @Dawson1917 in either case, is it REALLY that hard to take apart the window rather than smash it down? You don't have to destroy everything you know

Empowering powergamers with another reason to loot tools roundstart is something nobody wants to do.
t. powergamer, etc

ATH1909 commented:
Uh, how are we supposed to break out of permajail (without begging other people to let you out) now?

ATH1909 commented:
This change is a very, very bad idea, IMO. Reinforced windows already take a while to break through; I don't see why this change is necessary, and it looks like it'll encourage more bad behavior than it'll prevent.

anconfuzedrock commented:
This is basically making every window on the station plasmaglass. why not just make particularly important windows plasmaglass instead of doing this?

ATH1909 commented:
As far as I can tell, this change is only hurting newer players/non-powergamers (who don't know how to disassemble windows (instead of bashing them down) or who don't tide for tools roundstart) while leaving tiders at almost the same (B&E) power level that they were at before.

actioninja commented:
> 90% of station windows now requiring a fireaxe or esword to break through is ridiculous considering you're most likely to be smashing a window into chemistry for meds/cures in the absence of chemists or breaking into RnD because they are literal troglodytes who have neither researched nor upgraded anything on the station

congratulations, you are literally illustrating the exact reason why I want to see this change and are actively arguing against yourself.

ATH1909 commented:
In other words, this PR isn't actually making anything any more secure (anyone who'd be inclined to break into places in non-dire situations almost certainly always carries tools on their person)... but IS adding mechanics that could cause a lot of bullshit deaths/accidental GBJs.

actioninja commented:
It's part of some larger plans.
I could make them harder to deconstruct in this PR as well if you really think it's not going to make enough of a difference.

ATH1909 commented:
> It's part of some larger plans.
> I could make them harder to deconstruct in this PR as well if you really think it's not going to make enough of a difference.

Why are you trying to merge this PR before making the map modifications that this PR is supposed to work/not horribly break stuff with?

actioninja commented:
Because we don't know what those things that are going to be horribly broken are, or let alone if it's even going to happen.

ATH1909 commented:
And if this is supposed to be a part of a larger mapping change plan... what about the downstream servers who aren't using our maps?

actioninja commented:
They can deal with it themselves, if they even want to merge this.
I would prefer if you actually talked about real issues you had instead of feigning concern for downstreams you neither play on nor understand the ecosystem and pipeline of.

ATH1909 commented:
Why is people being able to break into each other's departments without tools a bad thing? Generally, if someone's trying to break into someone's department (and are willing to spend the time that it takes to do so without tools), they usually have a pretty darn good reason to do so.

actioninja commented:
As someone who breaks into departments a lot, no, they generally do not have a good reason.
Also please read the actual PR body before making comments.

ATH1909 commented:
I *did* read that PR body, but I don't agree with some of the points/evidence you raised in it.

Dawson1917 commented:
> 
> 
> > 90% of station windows now requiring a fireaxe or esword to break through is ridiculous considering you're most likely to be smashing a window into chemistry for meds/cures in the absence of chemists or breaking into RnD because they are literal troglodytes who have neither researched nor upgraded anything on the station
> 
> congratulations, you are literally illustrating the exact reason why I want to see this change and are actively arguing against yourself.

Some players are incompetent. Some players are absent. Some players are actual dickheads who won't do their jobs even if you ask nicely. When science holds up the entirety of the station, people are going to break in. When chemistry insists on making meth and lube bombs instead of healing people during crises, people are going to break in.  That is just how things are and making it more difficult only worsens matters. Windows shouldn't be coated in depleted uranium to prevent all these horrible psychopaths from fixing station problems.

actioninja commented:
"unga dunga man not play department way I want so i do self" is exactly the attitude this is an effort to make more difficult.

ATH1909 commented:
> "unga dunga man not play department way I want so i do self" is exactly the attitude this is an effort to make more difficult.

But WHY do you want to discourage that attitude? In other words, why is it a problem that if a department is grossly incompetent (possibly to the point of hindering the rest of the station), experienced people from other departments can easily break in and fix what the incompetent department is doing wrong?

ATH1909 commented:
As someone else mentioned above, why not just put reinforced plasmaglass windows in the sensitive places that you want to protect?

actioninja commented:
Because "this person isn't doing things the way I want them to so I'm going to break in and do it myself" is a shitty attitude that actively lowers the standard of interaction and sanity on the server.
This isn't about reinforcing sensitive areas, this is about making the station as a whole less of a playground for those who want to fuck shit up.

Dawson1917 commented:
> 
> 
> "unga dunga man not play department way I want so i do self" is exactly the attitude this is an effort to make more difficult.

"Fuck off c***, I'm not going to upgrade anything on the station. idgaf if people are dead and cloning takes an hour or chemistry's dispensers have no energy. Because I'm in science this place belongs to ME and I don't have to listen to you."
This is a frequent response for when I ask science if they can upgrade medbay or cargo, or plop an RPED with parts on the table. Similar deal with engis who refuse to set up power or fix breaches. People negatively impacting the round of everybody else on the server shouldn't be protected by turning their workplace into a fortress, they should be told to log off while other people smash down a window and fix the problem themselves because they refuse to.

ATH1909 commented:
Dawson's post made me realize something:

You claim that you're trying undo the fortress-ification of various departments with shutters and the like... yet this change basically turns every reinforced window into a shutter (unless you don't alter the deconstruction steps of reinforced windows, in which case it basically does nothing except inconvenience newer/non-tiding players).

actioninja commented:
Griefing shitheads who smash windows because the department isn't playing to their standard or just to steal shit are significantly more of a problem than people flat out refusing to do their job.
You can whine about it as much as you want, but this is going to continue going in circles. I don't care what you think if your entire complaint is that now it's harder to be a shithead who breaks into places.

Dawson1917 commented:
> 
> 
> Griefing shitheads who smash windows because the department isn't playing to their standard or just to steal shit are significantly more of a problem than people flat out refusing to do their job.
> You can whine about it as much as you want, but this is going to continue going in circles. I don't care what you think if your entire complaint is that now it's harder to be a shithead who breaks into places.

Someone intent on griefing will fetch a toolbelt and make their way around any convoluted deconstruction steps that might be added to reinforced windows. Or, god forbid, they'll hack one door to fetch the fireaxe from atmos or hack two doors to fetch the fireaxe from bridge and go nuts while everyone else suffers the moment they need to get out of a room. Or they'll... hack inside and be where they were before r-window buff. 
God, imagine the suffering of people thrown down disposals (or who try to escape a trapped r-window room through disposals) as they fruitlessly bang against the r-windows unable to escape GBJ until someone answers their very annoying begging over comms.

I see far more negligent players in departments than players wanting to smash windows 4no raisin. The 60 other players on the server shouldn't suffer because one person is being a chud.

Dawson1917 commented:
This just isn't the solution to the problem of "people break windows with little hesitation when they need/want to get inside somewhere" at all. Making windows invincible is bad and screws over more players than it could ever benefit. 
Plasmaglass already serves the purpose of super-reinforced glass that can absolutely tank anything you throw at it, and is very easily produceable. This PR actually makes windows more powerful than plasmaglass currently is with zero downsides, which is some real powercreep and/or real hugbox. 
What ever happened to engineers _interacting_ with people (particularly sec and chem and RnD) to reinforce their windows with plasmaglass so they'd get cool extremely durable windows and get to laugh at anyone who tries breaking in? The novelty is gone with every window to just about every department being untouchable from roundstart. Upgrading to plasma glass now will make your windows immune to literally... four extra weapons in the game.

optimumtact commented:
Do you see this 
![](https://file.house/Zsxx.png)
use it please

ATH1909 commented:
Just gonna copy+paste something I posted in Discord earlier here:

> if you want to nerf B&E, do it right:
> *increase the deconstruction times AND health pools of reinforced windows, but don't give them lame "you need this much force to damage this object" thresholds, since all those would really do is fuck over perma prisoners
> *increase the deconstruction times of walls so that walls aren't easier/faster to deconstruct than reinforced windows
> bam, done

and if you DO want to fuck over perma prisoners, just add the damage cutoff threshold to plasmaglass windows and replace the perma windows with plasmaglass windows

Shapsy commented:
But if all they need to get around those windows is a screwdriver and a crowbar, can't a dedicated shithead just get tools, pull the window out, then still loot the department, leaving only people who would try smashing a window in an emergency because they didn't powergame tools roundstart out of pocket? Why can't security deal with the person vandalizing and breaking in, their job isn't just to be hunting antags

81Denton commented:
I think that this will require some changes for edge cases like being stuck on a depowered arrivals shuttle; during European hours there are often no admins on non-Sybil servers that could help players out.

If this gets merged, I'll have a look at the integrity of regular glass windows. Right now pretty much every single window is an r-window because regular glass is so fragile; I'd like to make that more sturdy and swap r-windows in public areas for regular glass. That way we'll have more visual variety and can probably solve some GBJ issues.

CPTANT commented:
This shouldn't get merged because it creates lots of issues while trying to solve a non-problem.

optimumtact commented:
if you're just repeating existing points, or making contentless comments, stop now, or I will begin issuing timeouts as per usual.

zxaber commented:
Out of scope here, but if you ask me, airlocks (or airlock pairs) ought to be powered by a connected shuttle if their APCs are out. It would fix the issue with spawning players being stuck altogether.

Cyprex commented:
@ATH1909 Complaining because an oversight(being able to break out of perma) is fixed as a side effect of a PR is incredibly poor, please try to stay civil.

Swagile commented:
The issue of depowerment + EVA issues at Arrivals can be fixed by moving the 4 north / south lockers near Arrivals INSIDE of arrivals, and put 2 emergency toolboxes inside of arrivals shuttle; both on a table near the medkit in arrivals shuttle.

Ill probably do this myself later if I can figure out how to map.

nemvar commented on code/game/objects/structures/window.dm:
```diff
@@ -442,7 +441,7 @@
 	heat_resistance = 50000
 	armor = list("melee" = 85, "bullet" = 20, "laser" = 0, "energy" = 0, "bomb" = 60, "bio" = 100, "rad" = 100, "fire" = 99, "acid" = 100)
```
This has less melee armor than the other two. Oversight or is there some reason behind it?

nicbn commented:
The arrivals issue could be solved by making the airlocks be magically powered when the shuttle is docked, lasting some seconds after it departs so that the airlocks can close as well.

ATH1909 commented:
If new!reinforced windows are going to have 150 health and 90 melee armor, why do they also need a damage negation threshold? A <11 force weapon will take AGES to destroy a new!reinforced window anyway; I don't see why they need to be outright immune to them (except to confuse new players and screw over permabrig prisoners).

Swiftkor commented:
I see where he’s going with this. There is nothing more frustrating than trying to treat someone as a medical doctor who refuses your help and breaks into med storage because “me can do it better” People who do that can really just fuck off. Breaking into a department via a window makes everything look like shit and makes a poor engineer have to clean up after you later. If people in a department are absent and not able to help you should ask the AI to let you in. This problem might be somewhat fixed if we made request consoles relevant

CPTANT commented:
> 
> 
> I see where he’s going with this. There is nothing more frustrating than trying to treat someone as a medical doctor who refuses your help and breaks into med storage because “me can do it better” People who do that can really just fuck off. Breaking into a department via a window makes everything look like shit and makes a poor engineer have to clean up after you later. If people in a department are absent and not able to help you should ask the AI to let you in. This problem might be somewhat fixed if we made request consoles relevant

Then kick them out. Don't turn the entire station into a bunch of segregated zones that are impossible to break in or out off during an emergency or fight. Too much options are already gated behind having tools/RCD or not, this just loads yet more power on those tools.

>Breaking into a department via a window makes everything look like shit

The station going to shit is basically the premise of the game.

Dawson1917 commented:
Someone wanting to bash down a window to enter a room will use a toolbox/similar. Someone wanting to escape a room in an emergency will use whatever they have on hand - probably their oxygen tank. The latter is trapped for good.

2 minutes of toolbox bashing for one window is excessive. As long as deconstruction is quicker, aiming for 1 minute of bashing is much better, any more is obnoxious in play.

RedFoxIV commented:
were the wires not enough for you?

SkeletalElite commented:
This is much better than the old change, although after this get merged some windows should be swapped to regular windows instead of reinforced. Maybe regular windows should get a small buff to be more on the level of old reinforced windows?

YakumoChen commented:
Does a toolbox even do more than 11 damage?

CPTANT commented:
How about a video comparing how long it takes to bash down a window with these values?

Dawson1917 commented:
> 
> 
> Does a toolbox even do more than 11 damage?

Toolboxes do 12 damage

actioninja commented:
> How about a videos showing how long it actually takes to bash down a window with these values?

around 2 minutes.

pireamaineach commented:
This makes walls easier to get through than windows.

pireamaineach commented:
This basically just makes breaking windows fucking stupid in favour of deconstructing. It's not making windows harder to get through, it's removing options.

Whoneedspacee commented:
if you can deconstruct a wall you can probably deconstruct a window

pireamaineach commented:
Requires less tools. Also I covered deconstruction and why this PR in the end just removes options for getting through windows but Oranges told me to use edit and instead of waiting for me to copy he just deleted the message.

CPTANT commented:
> 
> 
> > How about a videos showing how long it actually takes to bash down a window with these values?
> 
> around 2 minutes.

That's ages. You can resist out of a bucklecuff in less time.

optimumtact commented:
And how does that have any relevance?

pireamaineach commented:
Because 2 minutes to break down a window in a station that can easily be traversed in 30 seconds is a massive, unneeded sec buff. Not because they can do their job better, but because their job is easier.

optimumtact commented:
Except I don't agree with that position, and neither it appears, do many other people

CPTANT commented:
> 
> 
> And how does that have any relevance?

It puts the time into perspective. It means security bucklecuffing you to a bed is less of a hindrance than having a single window between you and where you need to go. If you have to hit something 150 times most people won't even realize they are doing damage in the first place.

pireamaineach commented:
We both know that public opinion is irrelevant, and yet you bring it up on end when it agrees with you and disregard it when it doesn't.

optimumtact commented:
If public opinion was irrelevant warops would not be in the game, either way, this is offtopic so take it to the forums or discord.

CPTANT commented:
I also doubt the math that says it is going to take 2 minutes. I just tested it and under current conditions it take 15 seconds with a toolbox (locally hosted so it runs even faster) to smash an reinforced window. This update decreases melee damage by a factor of 5 and increases health by a factor of 3.  Which makes me believe actual time would be closer to 3 minutes 45 seconds.

An emergency tank now takes 45 seconds, good luck ever smashing a window with if if you are stuck.

pireamaineach commented:
Damage thresholds, tanks no longer work at all.

CPTANT commented:
even worse.

Swagile commented:
I don't really care about this PR easier for sec, im more concerned that this update makes it much harder for normal crew who don't powergame every round just to traverse the station in times of emergency.

You shouldn't be punished as a Medical Doctor for not rushing tool storage every round and end up dying to a plasma fire because r-windows can't be broken reliably by normal station weapons.

This will just serve to boost powergamers and experienced grey tiders arsenal while making normal crew even worse until a lot of windows are made into normal windows in return.

Either way, sec's problems aren't with normal tiders who break windows, its always the experienced players who are bored that are the problem, and they will know to take tools anyway.

pireamaineach commented:
Okay now not even tools are an option for quick WINDOW takedowns.

actioninja committed:
reinforced reinforced windows

actually 30 resist whoops

compiles

bit different of numbers

almost finished dcon stuff

new decon

examine messages

Whoneedspacee commented:
when do you ever need to break a window in medical to escape a plasma fire. if you don’t have a crowbar, sometimes you die anyways because of firelocks as well.

you are completely exaggerating how this will screw people over.

PKPenguin321 commented:
A soft damage cap would probably be better than a hard one to prevent people from getting stuck (damage that's 10 or less is divided by 4 and rounded down?).

optimumtact commented:
102 comments over windows :agony:

PKPenguin321 commented:
10000 reddit posts over oranges 🍊

actioninja commented:
Yep.

actioninja commented:
> I also doubt the math that says it is going to take 2 minutes. I just tested it and under current conditions it take 15 seconds with a toolbox (locally hosted so it runs even faster) to smash an reinforced window. This update decreases melee damage by a factor of 5 and increases health by a factor of 3. Which makes me believe actual time would be closer to 3 minutes 45 seconds.

That's not how math works.

GravitronBlue commented:
By now I think it should've be discussed why electric grilles aren't doing their job of protecting the windows of secure areas against break-ins, which seems to already be the present mechanism for the goal that this PR is attempting to reestablish.

For current feedback my thoughts are that adding armor deflection to reinforced glass and plasmaglass is fine and consistent with windoors, but buffing glass integrity and armor values is likely to have the biggest impact ingame and create an odd comparison right now between the difficulty of breaking windows vs breaking walls.

Regardless even with these changes there are numerous other ways to break into places, some methods immediately at the start of the round include
- cargo using disposal deliveries to send each other nearly anywhere else on the station
- chemists making thermite capable of melting rwalls
- engineering RCDs

and later:
- assistants with RCDs
- sonic jackhammers
- hulk from genetics
- handheld portal guns
- scientists stealing everything in range of RnD via telescience

Would suggest test-merging the PR for live testing at some point and seeing what it accomplishes.

Also I'd like to point this out:
> "This is part of some plans to make departments more secure overall..."
#45651 
![](https://user-images.githubusercontent.com/5194834/62287085-f89b1b80-b40d-11e9-8492-a2cd24c12234.png)
I'm sure parts of this PR are genuine, but if stuff like that is the plan it looks pretty biased towards just being another assistant nerf.

actioninja commented:
that pr is unrelated to the goals of making breaking and entering more difficult, I just hate that meta has so much bullshit free loot in the same places every round.

The other changes will likely be:
walls take longer to cut open, but retain the simple deconstruction
RCD is less easily availible/a no deconstruct version of RCD is as availible as now but the decon one is engineering only
hacking is reworked to be more challenging and not just hacking one door and maint and have ez fast pass for life

CPTANT commented:
>Same math for current windows:
>100 health, 12 damage, 50 armor
>100 / (12 * 0.5) = 16.6 or 17 seconds, which is close to what your testing showed. The 0.8 vs 1 click delay actually adds up a lot in practice from what I've found.

But current windows have 50 health.

actioninja commented:
I don't know why I just edited your comment instead of posting a new one, botched the reply maybe?
fulltiles have 100, directionals have 50

CPTANT commented:
> 
> 
> I don't know why I just edited your comment instead of posting a new one, botched the reply maybe?
> fulltiles have 100, directionals have 50

Ohw that actually explains a lot.

Tlaltecuhtli commented:
merge this so i can grab a rwindow stack and fill halls of unbreakable windows

actioninja committed:
proper values

actioninja committed:
removes accidental change

actioninja committed:
no repath edition

Whoneedspacee commented:
you could be annoying right now by making walls i don’t understand your point.

all you’re gonna do is get someone with an rcd to decon all your shit.

Akrilla commented:
I think everyone seriously overhypes the effect things will have on the game - exactly what @Whoneedspacee said - there's nothing stopping people currently from literally just building walls currently.

actioninja committed:
functional

Dorsisdwarf commented on code/game/objects/structures/window.dm:
```diff
@@ -511,16 +631,18 @@
 
 /obj/structure/window/reinforced/fulltile/unanchored
 	anchored = FALSE
+	state = WINDOW_OUT_OF_FRAME
```
Would it be possible to rename WINDOW_OUT_OF_FRAME to WINDOW_UNANCHORED, to avoid confusion with the new RWINDOW_OUT_OF_FRAME, which is 2 stages ahead of this stage?

Other than that couldn't find anything obviously fucked.

actioninja replied:
out of frame isn't the same as unanchored, they're related but anchoring is actually the first construction step and it's separate.

Dorsisdwarf commented on code/game/objects/structures/window.dm:
```diff
@@ -439,11 +495,74 @@
+				user.visible_message("<span class='notice'>[user] starts unfastening \the [src] from the frame...</span>",
+					"<span class='notice'>You start unfastening the bolts from the frame...</span>")
+				if(I.use_tool(src, user, 50, volume = 50))
+					to_chat(user, "<span class='notice'>You unscrew the bolts the frame and the window pops loose.</span>")
```
Unscrew the bolts FROM the frame

actioninja committed:
unnessecary code removal

Dorsisdwarf commented on code/game/objects/structures/window.dm:
```diff
@@ -382,12 +369,81 @@
+				user.visible_message("<span class='notice'>[user] starts unfastening \the [src] from the frame...</span>",
+					"<span class='notice'>You start unfastening the bolts from the frame...</span>")
+				if(I.use_tool(src, user, 50, volume = 50))
+					to_chat(user, "<span class='notice'>You unscrew the bolts the frame and the window pops loose.</span>")
```
FROM the frame here too

actioninja committed:
small cleanup

actioninja committed:
typo fix

CRITAWAKETS commented:
Maybe make regular windows a tiny bit stronger than r-windows were beforehand and replace areas that should not have stupidly strong windows with the new windows?

Also, titanium glass and plastitanium windows arent affected by this PR at all despite being very obviously tougher than windows made of iron and glass. Did you just forget or are you that lazy?

actioninja commented:
This is probably what's going to be done but overplanning to changes before they are even tested is a bad idea.

titanium and plastitanium windows are so uncommon as to practically not exist, considering that I forgot they did while making this. I'll just give them the same values.

Cyprex commented:
I'd really prefer if these were properly tiered, glass->rglass->plastitanium glass->plastitanium r-glass(does this exist?)->plasmaglass->plasmaglass, with each tier stronger than the preceding one.

YakumoChen commented:
Shuttle windows are effectively just reinforced windows with different metal. Not sure about infiltrator windows but given nukeops meta is "their shuttle is not easy to bomb" I'd assume it's around the strength of rplasglass

bigfatbananacyclops commented:
i wouldn't like to see engineering get more often raided for tools really, there's so many assistants and still the tools in storage isn't enough, people constantly breaks into engineering for insulated gloves (like me).

strong windows should be for windows adjacent/exposed to space, its dumb that a window protecting the station from breaches of space is so weak. still curious to see how this goes.

optimumtact commented:
I want to give this a go

Shirbu commented:
What would it take to Also apply this to the construction and SECURING of rglass? Currently, it is still the same screwdriver crowbar screwdriver to install them it appears

ShadowDimentio commented:
Classic oranges pushing terrible change that nobody wanted and ruins the game.

Now we have people flooding hallways with 3-3 rwindow cubes that are suddenly as unbreakable as fucking rwalls. Good job.

CPTANT commented:
I am actually surprised about half the people though hallways full of almost unbreakable windows was a good idea.

kopoba commented:
stunbaton not breaking reinforced window its make me sad can we fix this by nerfing window by 1 damage or buff stunbaton by 1 damage?

PKPenguin321 commented:
Refer to forum threads instead of necroing this PR please
